src/dateUtils.js

```diff
@@ -0,0 +1,42 @@
+const { DateTime, Duration } = require('luxon');
+
+const toDate = val => {
+  if (DateTime.isDateTime(val)){
+    return val;
+  } else if (typeof val === 'number'){
+    return DateTime.fromMillis(val);
+  } else if (typeof val === 'object' && val.getMonth && typeof val.getMonth === 'function'){
+    return DateTime.fromJSDate(val);
+  } else if (typeof val === 'string' && val.match(/^[0-9]{4}-[0-9]{2}-[0-9]{2}$/)){
+    return DateTime.fromISO(val);
+  } else if (typeof val === 'string' && val.match(/^[0-9]{2}\/[0-9]{2}\/[0-9]{4}$/)){
```

Comment:
Can we check `typeof val === 'string'` once and then match the regex in nested if?
e.g.
```js
else if (typeof val === 'string') {
  if(val.match(...)) {
  ...
  }
  else if(val.match(...)) {
  ...
  }
}
```

Reply:
I added the second format to match the date format [here](https://github.com/medic/medic-conf-test-harness/blob/4cbead1dfb99cd157aeb6cad1c76fa250d5c8dc6/test/harness.spec.js#L53), the two checks can be nested

Reply:
Updated